packages/types/src/primitive/Method.ts

```diff
@@ -86,21 +119,42 @@ export default class Method extends Struct implements IMethod {
    * @param _meta - Metadata to use, so that `injectMethods` lookup is not
    * necessary.
    */
-  private static decodeMethod (value: DecodedMethod | Uint8Array | string = new Uint8Array(), _meta?: FunctionMetadataV5): DecodedMethod {
+  private static decodeMethod (value: DecodedMethod | Uint8Array | string = new Uint8Array(), _meta?: MetaLike): DecodedMethod {
+
+    const unwrapMeta = (callIndex: Uint8Array, meta?: MetaLike): {
+      meta: FunctionMetadataV5;
+      method: string;
+      section: string;
+    } => {
+      if (meta instanceof FunctionMetadataV5) {
+        return { meta } as MethodFunction;
+      }
+
+      const methodFunction = (!meta
+        ? Method.findFunction(callIndex) // Global lookup
+        : meta instanceof Metadata
+          ? Method.findByCallIndex(callIndex, meta) // meta is the runtime metadata
+          : meta) as MethodFunction;
+
+      return { meta: methodFunction.meta, method: methodFunction.method, section: methodFunction.section };
+    };
+
     if (isHex(value)) {
       return Method.decodeMethod(hexToU8a(value), _meta);
     } else if (isU8a(value)) {
       // The first 2 bytes are the callIndex
       const callIndex = value.subarray(0, 2);
 
       // Find metadata with callIndex
-      const meta = _meta || Method.findFunction(callIndex).meta;
+      const { meta, method, section } = unwrapMeta(callIndex, _meta);
 
       return {
         args: value.subarray(2),
         argsDef: Method.getArgsDef(meta),
         callIndex,
-        meta
+        meta,
+        method,
+        section
       };
     } else if (isObject(value) && value.callIndex && value.args) {
```

Comment:
... and add another case here:

```javascript
else if (value instanceof Method) {
  return Method.{args,argsDef,callIndex,meta,method,section}
}
```

So that the logic "ignore options.meta" is done here, not in constructors of all other types that need Method